### PR TITLE
[SPARK-43525][BUILD] Import `scala.collection` instead of `collection`

### PR DESCRIPTION
**Pull request by panbingkun**
### What changes were proposed in this pull request?
- The pr aims to add check rules, importing in `collection`(instead with `scala.collection`) format is not allowed.
- Adjust the code import according to the above rules.

### Why are the changes needed?
I found that some developers sometimes write `collection.JavaConverters._` when import `JavaConverters._`, while others write `scala.JavaConverters._`
Actually, they all belong to the `scala group`, so it is necessary for us to specify a clearer rule to make the code style more consistent

### Does this PR introduce _any_ user-facing change?
No.

### How was this patch tested?
Pass GA.

### CODE REVIEW

**Comment by panbingkun** [comment]
cc @srowen

File: connector/kinesis-asl/src/main/scala/org/apache/spark/streaming/kinesis/KinesisInputDStream.scala [hunk 1]
```diff
@@ -17,9 +17,9 @@
 
 package org.apache.spark.streaming.kinesis
 
+import collection.JavaConverters._
```
Comment (HyukjinKwon) [inline connector/kinesis-asl/src/main/scala/org/apache/spark/streaming/kinesis/KinesisInputDStream.scala:20]: Can we just write it as scala.collection.*, and disallow other cases? Would be great to double check Scala guidance for this case.

Comment (LuciferYang) [reply connector/kinesis-asl/src/main/scala/org/apache/spark/streaming/kinesis/KinesisInputDStream.scala]: Is it feasible to add a `IllegalImportsChecker` item to `scalastyle-config.xml`  like:

```
  <check level="error" class="org.scalastyle.scalariform.IllegalImportsChecker" enabled="true">
    <parameters><parameter name="illegalImports"><![CDATA[collection.]]></parameter></parameters>
    <customMessage>Please use scala.collection instead.</customMessage>
  </check>
```

Comment (panbingkun) [reply connector/kinesis-asl/src/main/scala/org/apache/spark/streaming/kinesis/KinesisInputDStream.scala]: 1.I have checked the specifications of scala and there are no clear regulations.
2.At the same time, I searched the scala source code and found that in the `2.12.17` version,
the number of import writing methods in the form of `import scala.collection.JavaConverters._` is 37, while the number of import writing methods in the form of `import collection.JavaConverters._` is 3.

So, it is reasonable for us to prohibit importing in the form of `import collection.JavaConverters._` here.

3.In the `2.13.8` version, the number of import writing methods in the form of `import scala.collection.JavaConverters._` is 3, while the number of import writing methods in the form of `import collection.JavaConverters._` is 1.
According to official documents, in the 2.13 version, `collection.JavaConverters` has already been deprecated, so we need to consider this issue in the future.
image
link: https://docs.scala-lang.org/overviews/core/collections-migration-213.html#deprecated-things-in-212-that-have-been-removed-in-213

Comment (panbingkun) [reply connector/kinesis-asl/src/main/scala/org/apache/spark/streaming/kinesis/KinesisInputDStream.scala]: > Is it feasible to add a `IllegalImportsChecker` item to `scalastyle-config.xml` like:
> 
> ```
>   <check level="error" class="org.scalastyle.scalariform.IllegalImportsChecker" enabled="true">
>     <parameters><parameter name="illegalImports"><![CDATA[collection.]]></parameter></parameters>
>     <customMessage>Please use scala.collection instead.</customMessage>
>   </check>
> ```

Yes, that's right. Currently, this solution is more reasonable.

**Review by srowen — APPROVED** [review]

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM.

**Comment by dongjoon-hyun** [comment]
Could you re-trigger the failed pipeline, @panbingkun ?

**Review by HyukjinKwon — APPROVED** [review]

**Comment by HyukjinKwon** [comment]
Merged to master.

**Comment by LuciferYang** [comment]
late LGTM

**Pull request closed without merging**